blog/02-02-2023-connect-and-interact-with-celo-using-web3onboard-library/connect-and-interact-with-celo-using-web3onboard-library.md

```diff
@@ -0,0 +1,1106 @@
+    await disconnect().then(() => {
+      setAuthentication(false);
+    });
+  }
+```
+![image](images/14.png)
+
+Your `pages/index.tsx` should look like this.
+
+```bash
+import React from "react";
+import App from "../components/App";
+import LandingPage from "../components/LandingPage";
+import { useConnectWallet, useSetChain, useWallets } from "@web3-onboard/react";
+import { web3Onboard } from "@/components/apis/web3Onboard/setup";
+
+export default function Home() {
+  const [isUserAuthenticated, setAuthentication] = React.useState<boolean>(false);
+
+  const [, connect, , disconnect] = useConnectWallet();
+
+  async function handleConnect() {
+    if (isUserAuthenticated) return;
+    try {
+      await web3Onboard?.connectWallet().then(walletStates => {
+        if (walletStates.length) {
+          setAuthentication(true);
+        }
+      });
+    } catch (error) {
+      console.log("Connect Error", error);
+    }
+  }
+
+  async function logout() {
+    if (!isUserAuthenticated) throw new Error("User not authenticated");
+    await disconnect().then(() => {
+      setAuthentication(false);
+    });
+  }
+
+  return <>{!isUserAuthenticated ? <LandingPage isUserAuthenticated={isUserAuthenticated} handleConnect={handleConnect} /> : <App logout={logout} reconnect={connect} />}</>;
+}
+
+```
+
+## components/ConnectButton.tsx
+
+Make a new file name `ConnectButton.tsx` inside `components` folder. We are simply separating the 'Button' component as against what we currently have in `components/LandingPage.tsx`.
+
+Since we already initialized web3Onboard, we can use any of the `@web3-onboard/react` hook anywhere in the application. Since connecting users' wallet is done via an asynchronous function, the 'connecting' property returned by the 'useConnectWallet' hook helps us deactivate the connectButton when the connection has been invoked but not yet ready so users don't continiously click the button.
+
+```bash
+import React from "react";
+import { Button, Typography } from "@mui/material";
+import { ConnectButtonProp } from "../interfaces";
+import { useConnectWallet } from "@web3-onboard/react";
+import { green } from "@mui/material/colors";
+
+function ConnectButton(props: ConnectButtonProp) {
+  const [{ connecting }, ] = useConnectWallet();
+
+  const { connect } = props;
+
+  return (
+    <Button 
+      disabled={connecting} 
+      sx={{
+        width: '50%',
+        height: '120px',
+        border: '0.1em solid purple',
+        color: green[400],
+        borderRadius: '6px',
+        textAnchor: 'start',
+      }} 
+      variant="text" onClick={async() => await connect() }
+      className='connectButton'
+    >
+      <Typography variant={"h6"} >Connect Wallet</Typography>
+    </Button>
+  );
+}
+
+export default ConnectButton;
+
+```
+
+## components/LandingPage.tsx
+
+Import the ConnectButton into the `LandingPage.tsx` and the new code should look thus:
+
+```bash
```

Comment:
``` js?

Reply:
isn't this supposed to ```js?

Reply:
Please check the guidelines for review. The correct format is to use one '#' for the heading.